strawberryfields/backends/bosonicbackend/bosoniccircuit.py

```diff
@@ -803,6 +807,63 @@ def measure_dyne(self, covmat, modes, shots=1):
 
         return vals
 
+    def measure_threshold(self, modes):
+        r"""Performs photon number measurement on the given modes"""
+        if len(modes) == 1:
+            if self.active[modes[0]] is None:
+                raise ValueError("Cannot apply measurement, mode does not exist")
+
+            Idmat = self.hbar * np.eye(2) / 2
+            vacuum_fidelity = np.abs(self.fidelity_vacuum(modes))
+            measurement = np.random.choice((0, 1), p=[vacuum_fidelity, 1 - vacuum_fidelity])
+            samples = measurement
```

Comment:
I just changed the shape of the output samples (not putting it in an array) so that they match the shape of other types of measurements.